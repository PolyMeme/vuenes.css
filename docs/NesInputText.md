# nes-input-text 

## props 

- `id` ***String*** (*required*) 

  Define the id of the input 

- `label` ***String*** (*optional*) `default: Empty String` 

  Define the label text of the input 

- `placeholder` ***String*** (*optional*) `default: Empty String` 

  Define te placeholder of the input 

- `init-value` ***String*** (*optional*) `default: Empty String` 

  Define the init value of the input 

- `inline` ***Boolean*** (*optional*) `default: false` 

  Define if the Input is type inline by using the `is-inline` css class 

- `success` ***Boolean*** (*optional*) `default: false` 

  Define if the Input is type success by using the `is-success` css class 

- `warning` ***Boolean*** (*optional*) `default: false` 

  Define if the Input is type warning by using the `is-warning` css class 

- `error` ***Boolean*** (*optional*) `default: false` 

  Define if the Input is type error by using the `is-error` css class 

## data 

- `inputValue` 

**initial value:** `[object Object]` 

## computed properties 

- `inputClasses` 

  Calculate the css classes to be used for the input text depending of the props 

   **dependencies:** `success`, `warning`, `error` 


## events 

- `input` 

  Fired when someone does an input in the input text 

## methods 

- `emitInput()` 

  Method that emits the input event to the parent component 

