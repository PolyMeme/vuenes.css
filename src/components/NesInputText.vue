<template>
  <div
    :class="{'is-inline': inline}"
    class="nes-field"
  >
    <label
      v-if="label"
      :for="id"
    >
      {{ label }}
    </label>
    <input
      :id="id"
      v-model="inputValue"
      type="text"
      :class="inputClasses"
      class="nes-input"
      :placeholder="placeholder"
      @input="emitInput"
    >
  </div>
</template>

<script>
export default {
  props: {
    id: {
      type: String,
      required: true
    },
    label: {
      type: String,
      default: () => ''
    },
    placeholder: {
      type: String,
      default: () => ''
    },
    initValue: {
      type: String,
      default: () => ''
    },
    inline: Boolean,
    success: Boolean,
    warning: Boolean,
    error: Boolean
  },
  data () {
    return {
      inputValue: this.initValue
    }
  },
  computed: {
    inputClasses () {
      return {
        'is-success': this.success,
        'is-warning': this.warning,
        'is-error': this.error
      }
    }
  },
  methods: {
    emitInput () {
      this.$emit('input', this.inputValue)
    }
  }
}
</script>

<style lang="scss">
@import "~typeface-press-start-2p/index.css";
@import "~nes.css/scss/base/index";
@import "~nes.css/scss/form/inputs";
</style>
