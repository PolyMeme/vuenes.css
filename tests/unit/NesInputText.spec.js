import { mount } from '@vue/test-utils'
import NesInputText from '../../src/components/NesInputText.vue'

describe('NesInputText.vue', () => {
  const createInputWrapper = (propsData = {}) => {
    return mount(NesInputText, {
      propsData
    })
  }

  it('should render the component properly', () => {
    const wrapper = createInputWrapper({
      id: 'test-input'
    })

    expect(wrapper.find('div.nes-field>input[type="text"]#test-input.nes-input').exists()).toBe(true)
  })

  it('should print input with label', () => {
    const wrapper = createInputWrapper({
      id: 'test-input',
      label: 'the label!'
    })

    expect(wrapper.find('div.nes-field>label+input[type="text"]#test-input.nes-input').exists()).toBe(true)
    expect(wrapper.find('label').text()).toBe('the label!')
  })

  it('should render input with placeholder', () => {
    const wrapper = createInputWrapper({
      id: 'test-input',
      placeholder: 'this is the placeholder'
    })

    expect(wrapper.find('input#test-input').attributes().placeholder).toEqual('this is the placeholder')
  })

  it('should emit input event on input', () => {
    const wrapper = createInputWrapper({
      id: 'test-input'
    })

    // Trigger Click and Emit Three Times
    const selector = 'input.nes-input#test-input'
    wrapper.find(selector).trigger('input')
    wrapper.find(selector).trigger('input')
    wrapper.find(selector).trigger('input')

    expect(wrapper.emitted('input')).toBeTruthy()
    expect(wrapper.emitted('input').length).toBe(3)
  })

  it('should match snapshot', () => {
    const wrapper = createInputWrapper({
      id: 'test-input',
      label: 'This is a label',
      placeholder: 'Insert Text',
      inline: true,
      info: true
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
