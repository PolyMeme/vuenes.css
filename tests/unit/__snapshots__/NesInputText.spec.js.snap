// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesInputText.vue should match snapshot 1`] = `
<div class="field is-inline" info="true"><label for="test-input">
    This is a label
  </label> <input id="test-input" type="text" placeholder="Insert Text" class="input"></div>
`;
